app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt

```diff
@@ -0,0 +1,160 @@
+package com.github.se.travelpouch.model.activity
+
+import android.util.Log
+import com.github.se.travelpouch.model.Location
+import com.google.android.gms.tasks.Task
+import com.google.firebase.Firebase
+import com.google.firebase.Timestamp
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+class ActivityRepositoryFirebase(private val db: FirebaseFirestore) : ActivityRepository {
+
+    private val collectionPath = "activities"
+
+    /**
+     * This function returns an unused unique identifier for a new event.
+     *
+     * @return (String) : an unused unique identifier
+     */
+    override fun getNewUid(): String {
+        return db.collection(collectionPath).document().id
+    }
+
+    /**
+     * The initialisation function of the repository.
+     *
+     * @param (() -> Unit) : the function to apply when the authentication goes without a trouble
+     */
+    override fun init(onSuccess: () -> Unit) {
+        Firebase.auth.addAuthStateListener {
+            if (it.currentUser != null) {
+                onSuccess()
+            }
+        }
+    }
+
+    override fun updateActivity(
+        activity: Activity,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        Log.d("TravelRepositoryFirestore", "updateTravel")
+        performFirestoreOperation(
+            db.collection(collectionPath).document(activity.uid).set(activity), onSuccess, onFailure)
+    }
+
+    override fun deleteActivityById(
+        id: String,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        Log.d("ActivityRepositoryFirestore", "deleteTravelById")
+        performFirestoreOperation(
+            db.collection(collectionPath).document(id).delete(), onSuccess, onFailure)
+    }
+
+    /**
+     * This function retrieves all the events of a travel from the database. If the operation succeeds
+     * a function is applied on the list of events retrieved, otherwise a failure function is called.
+     *
+     * @param onSuccess (List<Event>) -> Unit): The function called when the retrieving of the events
+     *   went successfully.
+     * @param onFailure ((Exception) -> Unit): The function called when the retrieving of the events
+     *   fails.
+     */
+    override fun getActivity(onSuccess: (List<Activity>) -> Unit, onFailure: (Exception) -> Unit) {
```

Comment:
Rename methods: Consider renaming getActivity() to getAllActivities()

Reply:
Yes, it is more explicit.